### PR TITLE
Changing Style Guide for internal names (Issue #908)

### LINKED ISSUE #908: Private or Internal Names
The section on *Private or Internal Names* should be revised to recommend the use of the `internal` keyword rather than prefixing with `_`.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 5cc4b9d4-0214-5c5c-4528-20d345c87a2d
* Version Independent ID: 4e5ad251-dec1-1e76-7089-eb8862d24160
* Content: [Microsoft Q# Style Guide - Microsoft Quantum](https://docs.microsoft.com/en-us/quantum/contributing/style-guide?tabs=guidance)
* Content Source: [articles/contributing/style-guide.md](https://github.com/MicrosoftDocs/quantum-docs-pr/blob/live/articles/contributing/style-guide.md)
* Product: **quantum**
* GitHub Login: @cgranade
* Microsoft Alias: **chgranad**

### PR DESCRIPTION
**Pull request by geduardo**
This fixes #908.

### CODE REVIEW

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards

**A minimum Acrolinx score of 80 is required.**

Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology.

| Article | Acrolinx<br>score | Lowest<br>category<br>score | Spelling<br>issues | Scorecard | Processed |
|---------|:--------------:|:--------------------:|:------:|:---------:|:---------:|
| [articles/contributing/style-guide.md](https://github.com/MicrosoftDocs/quantum-docs-pr/blob/4f9c19c3efc0584dc87a22ccaa5d967c13f49227/articles/contributing/style-guide.md) | 81 |   | 2 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/5c72efa6-5b32-48b9-8d13-1b2bad98fe46_report.html) | [:white_check_mark:](# "Successfully processed") |

**More information about Acrolinx**

- [Install Acrolinx locally for VSCode](https://review.docs.microsoft.com/help/contribute/contribute-acrolinx-vscode)
- [False positives or issues](https://aka.ms/acrolinxbug)
- [Troubleshooting issues with Acrolinx](https://review.docs.microsoft.com/help/contribute/acrolinx-error-messages)

File: articles/contributing/style-guide.md [hunk 1]
```diff
@@ -208,23 +208,22 @@ We suggest:
...
 It is helpful to clearly indicate that this is the case when naming functions and operations so that accidental dependencies on internal-only code are made obvious.
-If an operation or function is not intended for direct use, but rather should be used by a matching callable which acts by partial application, consider using a name starting with `_` for the callable that is partially applied.
+If an operation or function is not intended for direct use, but rather should be used by a matching callable which acts by partial application, consider using a name starting with the `internal` keyword for the callable that is partially applied.
 
 # [Guidance](#tab/guidance)
 
 We suggest:
 
-- When a function, operation, or user-defined type is not a part of the public API for a Q# library or program, ensure that its name begins with a leading underscore (`_`).
+- When a function, operation, or user-defined type is not a part of the public API for a Q# library or program, ensure that its name begins with the leading keyword `internal`.
```
Comment (cgranade) [inline articles/contributing/style-guide.md:217]: Strictly speaking, the internal keyword isn't part of the name.
```suggestion
- When a function, operation, or user-defined type is not a part of the public API for a Q# library or program, ensure that it is marked as internal by placing the `internal` keyword before the `function`, `operation`, or `newtype` declaration.
```

File: articles/contributing/style-guide.md [hunk 2]
```diff
@@ -208,23 +208,22 @@ We suggest:
...
 
 # [Examples](#tab/examples)
 
 |   | Name | Description |
 |---|------|-------------|
-| ☒ | <s>`ApplyDecomposedOperation_`</s> | The underscore `_` should not appear at the end of the name. |
-| ☑ | `_ApplyDecomposedOperation` | The underscore `_` at the beginning clearly indicates that this operation is for internal use only. |
+| ☒ | <s>`_ApplyDecomposedOperation`</s> | Do not use an underscore `_` to indicate that this operation is for internal use only. |
+| ☑ | `internalApplyDecomposedOperation` | The `internal` keyword at the beginning clearly indicates that this operation is for internal use only. |
```
Comment (cgranade) [inline articles/contributing/style-guide.md:224]:
```suggestion
| ☒ | <s>`operation _ApplyDecomposedOperation`</s> | Do not use an underscore `_` to indicate that this operation is for internal use only. |
| ☑ | `internal operation ApplyDecomposedOperation` | The `internal` keyword at the beginning clearly indicates that this operation is for internal use only. |
```

File: articles/contributing/style-guide.md [hunk 3]
```diff
@@ -208,23 +208,22 @@ We suggest:
 
 In many cases, a name is intended strictly for use internal to a library or project, and is not a guaranteed part of the API offered by a library.
 It is helpful to clearly indicate that this is the case when naming functions and operations so that accidental dependencies on internal-only code are made obvious.
-If an operation or function is not intended for direct use, but rather should be used by a matching callable which acts by partial application, consider using a name starting with `_` for the callable that is partially applied.
+If an operation or function is not intended for direct use, but rather should be used by a matching callable which acts by partial application, consider using a name starting with the `internal` keyword for the callable that is partially applied.
```
Comment (msoeken) [inline articles/contributing/style-guide.md:211]:
```suggestion
If an operation or function is not intended for direct use, but rather should be used by a matching callable which acts by partial application, consider making the callable that is partially applied an internal operation using the `internal` modifier keyword.
```

File: articles/contributing/style-guide.md (hunk 1 above)
Comment (msoeken) [inline articles/contributing/style-guide.md:217]:
```suggestion
- When a function, operation, or user-defined type is not a part of the public API for a Q# library or program, ensure that the operation is declared internal using the `internal` modifier.
```

File: articles/contributing/style-guide.md (hunk 2 above)
Comment (msoeken) [inline articles/contributing/style-guide.md:224]:
```suggestion
| ☑ | `internal ApplyDecomposedOperation` | The `internal` keyword at the beginning clearly indicates that this operation is for internal use only. |
```

**Comment by msoeken** [comment]
It seems I made my review at the same time as @cgranade . I agree more with his suggestions, so please feel free to ignore my review.

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards

**A minimum Acrolinx score of 80 is required.**

Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology.

| Article | Acrolinx<br>score | Lowest<br>category<br>score | Spelling<br>issues | Scorecard | Processed |
|---------|:--------------:|:--------------------:|:------:|:---------:|:---------:|
| [articles/contributing/style-guide.md](https://github.com/MicrosoftDocs/quantum-docs-pr/blob/41c488e48d4deeed45f9d366bb214f9b8b962ed3/articles/contributing/style-guide.md) | 81 | 57 | 2 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/4e142742-fc35-4d79-bbbb-2046c9da8be2_report.html) | [:white_check_mark:](# "Successfully processed") |

**More information about Acrolinx**

- [Install Acrolinx locally for VSCode](https://review.docs.microsoft.com/help/contribute/contribute-acrolinx-vscode)
- [False positives or issues](https://aka.ms/acrolinxbug)
- [Troubleshooting issues with Acrolinx](https://review.docs.microsoft.com/help/contribute/acrolinx-error-messages)

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards

**A minimum Acrolinx score of 80 is required.**

Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology.

| Article | Acrolinx<br>score | Lowest<br>category<br>score | Spelling<br>issues | Scorecard | Processed |
|---------|:--------------:|:--------------------:|:------:|:---------:|:---------:|
| [articles/contributing/style-guide.md](https://github.com/MicrosoftDocs/quantum-docs-pr/blob/d5bf21e5e9ae630bdd292cf69582be3168ca303e/articles/contributing/style-guide.md) | 81 | 57 | 2 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/eaad71e2-4c9d-45d3-a27d-2e8bada56585_report.html) | [:white_check_mark:](# "Successfully processed") |

**More information about Acrolinx**

- [Install Acrolinx locally for VSCode](https://review.docs.microsoft.com/help/contribute/contribute-acrolinx-vscode)
- [False positives or issues](https://aka.ms/acrolinxbug)
- [Troubleshooting issues with Acrolinx](https://review.docs.microsoft.com/help/contribute/acrolinx-error-messages)

**Pull request merged**